### PR TITLE
Read one letter from db at a time

### PR DESCRIPTION
**Pull request by rkondratowicz**
Also make it easier to change the batch size for job run

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java [hunk 1]
```diff
@@ -51,73 +52,78 @@ public UploadLettersTask(
...
         logger.info("Started '{}' task", TASK_NAME);
 
-        List<Letter> lettersToUpload = repo.findFirst3ByStatus(LetterStatus.Created);
-        int counter = 0;
-
-        if (!lettersToUpload.isEmpty()) {
-            counter = ftp.runWith(sftpClient -> uploadLetters(lettersToUpload, sftpClient));
+        if (!availabilityChecker.isFtpAvailable(now(ZoneId.of(EUROPE_LONDON)).toLocalTime())) {
+            logger.info("Not processing '{}' task due to FTP downtime window", TASK_NAME);
+        } else {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: now with this move makes it even more sensible to move `started this task` to aop component which we already have specific for schedules. just a blarb, not take care right now

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java [hunk 2]
```diff
@@ -51,73 +52,78 @@ public UploadLettersTask(
...
+                if (letter.isPresent()) {
+                    boolean uploaded = processLetter(letter.get(), client);
+                    if (uploaded) {
+                        uploadCount++;
+                    }
                 } else {
-                    letter.setStatus(LetterStatus.Skipped);
+                    break;
                 }
+            }
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: Bah, java9 here would be very handy with `.takeWhile` function

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java [hunk 3]
```diff
@@ -51,73 +52,78 @@ public UploadLettersTask(
...
+        return ftp.runWith(client -> {
+            int uploadCount = 0;
+
+            for (int i = 0; i < BATCH_SIZE; i++) {
+                Optional<Letter> letter = repo.findFirstByStatusOrderByCreatedAtAsc(LetterStatus.Created);
+                if (letter.isPresent()) {
+                    boolean uploaded = processLetter(letter.get(), client);
+                    if (uploaded) {
+                        uploadCount++;
+                    }
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: We could just use `if` statement alone without assigning new var

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: soon ;)

**Pull request merged**